src/vision/status_quo/alan_finds_database_drops_hard.md

```diff
@@ -0,0 +1,84 @@
+
+
+## Searching for the Solution
+
+
+Alan tries to figure out what happened from the logs, but is unable to do so. Alan decides to attempt to recreate the scenario by doing extra load testing and trying to emulate the increased traffic. After some time, Alan gets lucky and is able to recreate the error, but he still doesn't know what could be the cause. Alan turns to the documentation for the `sqlx` crate to see if there are flags that might enable extra instrumentation but he can't find any (XXX check what crate offers).
+
+Alan, next, takes the following steps to determine the most likely source to explain the problem behavior in his application:
+
+1. Check the `sqlx` documentation for possible flags to explicitly close the connection but none are found.
+2. Examine the source code to figure out where the connection gets closed.
+3. Discover that the connection is closed through the destructor, Drop::drop only when it gets cleaned up aftering going out of scope.
+
+Alan realizes he must seek help from someone more experienced and knowledgable. He goes to his friend Barbara, and asks, "How can I get a log statement when the destructor runs?". Barbara tells him to wrap his database handle in another struct which has a destructor and to insert a log into the destructor for that struct we'll call `Foo`. Alan implements `Foo` and sees the new log when the destructor runs in the output. Now that the log works Alan runs program with full load in order to reproduce the problem. Alan analyzes the logs and notices that the new connection is being created and panicing before the destructor has run.
+
+Next, Alan seeks advice from the `sqlx` forum. He learns that the DropImpl for new connections spawns a task in order to close a handle, and therefore, it may not execute right away. His advisor states that Rust doesn't have a way to execute async operations in a destructor.
+
+Alan evalutes possible fixes to this problem. His first consideration is whether there is an explicit async method that will close the connection. He then realizes that when a user disconnects and the active future for that user is dropped, his call wouldn't run anyway. 
+
+```rust=
+let handle = ...
+logic(handle).await; // if user disconnects here, call to `handle.close()` will never execute
+handle.close();
+```
+
+## Finding the Solution
+
+Alan briefly considers rearchitecting his application in more extreme ways to retain use of async, but he gives up and seeks a more straight forward solution. He discovers `rusqlite`, a sychronous database library and adopts it. This requires some rearchitecting but solves the problem.
+
+## 🤔 Frequently Asked Questions
+
+*Here are some standard FAQ to get you started. Feel free to add more!*
+
+* **What are the morals of the story?**
```

Comment:
I just want to note here that the FAQ section should be filled out before we merge.

Reply:
Yes, that was understood, but I didn't want to add FAQs until other questions within the story were resolved.  Once the story is complete it will be easier to add FAQs that correctly reflect the details in this story.